src/routes/users.js

```diff
@@ -1,43 +1,183 @@
+  debug('Hello World! dec 2019 here I am ')
 
-router.get('test/', function(req,res,next){
-debug('Hello World! dec 2019 here I am ');
+  db.User.findAll({
+    attributes: ['lastName', 'email'],
+  }).then((users) => {
+    res.json(users)
+  })
+})
+
+
+router.post('/reset_password_email', function (req, res, next) {
+  //  user by email
+  db.User.findOne({
+    where: { email: req.body.email }
+  })
+    .then((user) => generateStoreSendToken(user))
+    .catch((err) => res.send(err))
+
+  // if user exist, generate and store token in User table
+  function generateStoreSendToken(user) {
+    if (user) {
+      return crypto.randomBytes(127, function (err, buffer) {
+        if (err) throw err
+        const token = buffer.toString('hex')
+        return db.User.update(
+          {
+            resetPasswordToken: token,
+            resetPasswordExpiry: Date.now() + oneDay
+          },
+          {
+            where: { id: user.id },
+            returning: true,
+          }
+        )
+          .then(([, updated]) => {
+            sendEmail(updated[0])
+          })
+          .catch((err) => res.send(err))
+      })
+    } else if (!user) {
+      return res.status(404).send('User not found')
+    }
+  }
+
+  // send email with token to the user
+  function sendEmail(user) {
+    const mailConfig = {
+      host: 'smtp.mailtrap.io',
+      port: 2525,
+      auth: {
+        user: process.env.NODEMAILER_USER,
+        pass: process.env.NODEMAILER_PASSWORD
+      },
+    }
+    const transporter = nodeMailer.createTransport(mailConfig)
+    const mailOptions = {
+      from: '"Info" <info@example.com>', // sender address
+
+      to: user.email, // list of receivers
+
+      subject: 'Reset Password', // Subject line
+      context: {
+        url:
+          'http://localhost:3000/reset_password?token=' +
+          user.resetPasswordToken,
+        name: user.firstName,
+      },
+      html: `<body>
+      <div>
+          <h3>Dear ${user.firstName},</h3>
+          <br>
+          <p>You requested for a password reset, kindly use this
+           <a href="http://reach-fronted.herokuapp.com/reset_password?token=${user.resetPassswordToken}">link</a>
+            to reset your password. This link is valid for one day.
+          </p>
+          <br>
+          <p>Cheers!</p>
+      </div>
+     
+  </body>`,
+    }
+
+    transporter.sendMail(mailOptions, (error, info) => {
+      if (!error) {
+        return res.json({
+          message: `Kindly check your email for further instructions`,
+        })
+      } else {
+        return res.status(500).json({
+          message:
+            'Oops there is an error...Please try again or contact your admin if the issue persist',
+        })
+      }
+    })
+  }
+})
+
+router.put('/reset_password', function (req, res, next) {
+  const { token, password1, password2 } = req.body
+  const bcryptedPassword = bcrypt.hashSync(password1, bcrypt.genSaltSync())
```

Comment:
You can do this after the check for missing/differing passwords.

Reply:
okay done:)